code/datums/jobs/job/job_exp.dm

```diff
@@ -198,26 +199,16 @@ GLOBAL_PROTECT(exp_to_update)
 			to_chat(src,"<span class='notice'>You got: [minutes] Admin EXP!</span>")
 
 	if(isliving(mob))
+		var/mob/living/living_mob = mob
 		if(mob.stat != DEAD)
-			var/rolefound = FALSE
 			play_records[EXP_TYPE_LIVING] += minutes
 			if(announce_changes)
 				to_chat(src,"<span class='notice'>You got: [minutes] Living EXP!</span>")
-			if(mob.mind?.assigned_role)
-				for(var/job in SSjob.name_occupations)
-					if(mob.mind.assigned_role == job)
-						rolefound = TRUE
-						play_records[job] += minutes
-						if(announce_changes)
-							to_chat(src,"<span class='notice'>You got: [minutes] [job] EXP!</span>")
-				if(!rolefound)
-					for(var/role in GLOB.exp_specialmap[EXP_TYPE_SPECIAL])
-						if(mob.mind.assigned_role == role)
-							rolefound = TRUE
-							play_records[role] += minutes
-							if(announce_changes)
-								to_chat(mob,"<span class='notice'>You got: [minutes] [role] EXP!</span>")
-			if(!rolefound)
+			if(living_mob.job)
+				play_records[living_mob.job.title] += minutes
+				if(announce_changes)
+					to_chat(src,"<span class='notice'>You got: [minutes] [living_mob.job] EXP!</span>")
```

Comment:
lol i didn't know we actually would announce these at some point :|

Reply:
Admins are not affected by it.

Reply:
Either way is this not spammy?

Reply:
It's not called anywhere with this variable on a true value, so I guess it's more for testing.